src/pybind/mgr/dashboard/rbd_iscsi.py

```diff
@@ -27,22 +27,23 @@ def _get(self):
                         }
                         daemons[server['hostname']] = daemon
 
-                    image = images.get(service['id'])
+                    image_id = service['id'].split(':')[-1]
+                    image = images.get(image_id)
                     if image is None:
                         image = {
-                            'id': service['id'],
+                            'id': image_id,
                             'pool_name': metadata['pool_name'],
                             'name': metadata['image_name'],
                             'optimized_paths': [],
                             'non_optimized_paths': []
                         }
-                        if status.get('lock_owner', 'false') == 'true':
-                            daemon['optimized_paths'] += 1
-                            image['optimized_paths'].append(server['hostname'])
-                        else:
-                            daemon['non_optimized_paths'] += 1
-                            image['non_optimized_paths'].append(server['hostname'])
-                        images[service['id']] = image
+                        images[image_id] = image
+                    if status.get('lock_owner', 'false') == 'true':
```

Comment:
Is the de-indent here intentional?  I'm guessing yes, because it seems reasonable that you'd want the optimized_paths stuff irrespective of the `image is None` check, but thought I'd double check.